client/extensions/woocommerce/state/sites/settings/email/test/reducer.js

```diff
@@ -0,0 +1,186 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST,
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT_FAILURE,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_API_KEY_SUBMIT,
+	WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+	WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_FAILURE,
+} from 'woocommerce/state/action-types';
+import reducer from 'woocommerce/state/sites/reducer';
+
+describe( 'reducer', () => {
+	test( 'should mark the settings request fetching', () => {
+		const siteId = 123;
+		const action = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST,
+			siteId,
+		};
+
+		const newSiteData = reducer( {}, action );
+		expect( newSiteData[ siteId ].settings.email.settingsRequest ).to.eql( true );
+	} );
+
+	test( 'should store data from the action', () => {
+		const siteId = 123;
+		const settings = { store_name: 'mystore' };
+		const action = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+			siteId,
+			settings,
+		};
+		const newState = reducer( {}, action );
+		expect( newState[ siteId ] ).to.exist;
+		expect( newState[ siteId ].settings ).to.exist;
+		expect( newState[ siteId ].settings.email.settings ).to.deep.equal( settings );
+	} );
+
+	test( 'should merge lists data into settigns and pick first as default if none is chosen', () => {
+		const siteId = 123;
+		const settings = { store_name: 'mystore' };
+		const lists = { a: 'list_a', b: 'list_b' };
+		const settingsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_SETTINGS_REQUEST_SUCCESS,
+			siteId,
+			settings,
+		};
+
+		const listsAction = {
+			type: WOOCOMMERCE_MAILCHIMP_LISTS_REQUEST_SUCCESS,
+			siteId,
+			lists,
+		};
+
+		const finalState = {
+			store_name: 'mystore',
+			mailchimp_lists: {
+				a: 'list_a',
+				b: 'list_b',
+			},
+			mailchimp_list: 'a',
+		};
+
+		const firstState = reducer( {}, settingsAction );
+		const updatedState = reducer( firstState, listsAction );
+		expect( updatedState[ siteId ] ).to.exist;
+		expect( updatedState[ siteId ].settings ).to.exist;
+		expect( updatedState[ siteId ].settings.email.settings ).to.deep.equal( finalState );
+	} );
+
+	test( 'should merge lists data into settigns and not update chosen list', () => {
```

Comment:
typo here for `settings`.

It might also help to clarify this test a bit `should merge lists data into settings and not update the chosen list when a list has been selected`.